src/Storages/MergeTree/GinIndexStore.cpp

```diff
@@ -511,6 +513,30 @@ void GinIndexStore::cancel() noexcept
         postings_file_stream->cancel();
 }
 
+String GinIndexStore::Statistics::toString() const
+{
+    return fmt::format(
+        "number of terms = {}, terms size = {}, metadata size = {}, bloom filter size = {}, dictionary size = {}, posting lists size = {}",
+        num_terms,
+        ReadableSize(current_size),
+        ReadableSize(metadata_file_size),
+        ReadableSize(bloom_filter_file_size),
+        ReadableSize(dictionary_file_size),
+        ReadableSize(posting_lists_file_size));
+}
+
+GinIndexStore::Statistics GinIndexStore::getStatistics()
+{
+    return {
+        .num_terms = current_postings.size(),
+        .current_size = current_size,
+        .metadata_file_size = metadata_file_stream ? metadata_file_stream->offset() : 0,
```

Comment:
Looking at src/IO/BufferBase.h, `offset()` is only the size of the working buffer. Function `count()` returns _all_ written bytes (including those where the working buffer was flushed already)

Reply:
ah yes it should be `.position()`.